### ISSUE TITLE
musl doesn't have sys_siglist

### ISSUE THREAD

**Issue by daurnimator** [wahern/lunix#9]
(I'm not using configure.ac, just compiling manually)
```
musl-gcc -O2 -fPIC -I/usr/include/lua5.1 -c src/unix.c -o src/unix.o -D_REENTRANT -D_THREAD_SAFE -D_GNU_SOURCE
src/unix.c: In function ‘unixL_strsignal’:
src/unix.c:2635:44: error: ‘sys_siglist’ undeclared (first use in this function)
  if (signo >= 0 && signo < NSIG && (info = sys_siglist[signo]))
```

musl doesn't have a `sys_siglist`. defaulting `HAVE_SYS_SIGLIST` to 1 seems wrong.
however it *does* have `strsignal`, but doesn't meet any of the requirements in the `#ifdef`:
```C
(USE_STRSIGNAL || __sun || GLIBC_PREREQ(0,0) || FREEBSD_PREREQ(8,1) || defined __APPLE__ || defined _AIX)
```

A sufficient check should just be `_POSIX_C_SOURCE >= 200809L`

**Comment by wahern** [comment]
Oops. This slipped through the cracks. I committed the fix.

The autoconf stuff isn't complete yet and shouldn't be used. But on Alpine Linux I _was_ using it, as musl libc was the impetus for adding autoconf support. Because the built-in checks were skipped and configure.ac is missing most checks, both HAVE_SYS_SIGLIST and HAVE_STRSIGNAL were undefined and evaluated to 0. So everything built fine for me in my periodic tests because I forgot to clean away the cached flags.

`_POSIX_C_SOURCE` is something the _application_ is supposed to define. glibc is an odd-ball in setting it when not yet defined. (AIX does, too, but not Solaris, musl, OS X, or any of the BSDs I tested.) I never set `_POSIX_C_SOURCE` because on many systems it has the effect of disabling many useful, often de facto portable, system extensions.
